pkg/river/vm/vm.go

```diff
@@ -85,16 +85,47 @@ func (vm *Evaluator) evaluateBlockOrBody(scope *Scope, assoc map[value.Value]ast
 		rv = rv.Addr()
 	}
 
+	return vm.evaluateBlockOrBodyDefault(scope, assoc, node, rv)
+}
+
+func (vm *Evaluator) evaluateBlockOrBodyDefault(scope *Scope, assoc map[value.Value]ast.Node, node ast.Node, rv reflect.Value) error {
```

Comment:
I'm on the fence about whether this makes sense as one longer method vs splitting it up like this; introducing four new intermediate methods might make the mental model of the call tree harder to all keep track of in (brain) memory at once.

Reply:
I might have an idea I can try here to get us the best of both worlds. It's a little more refactoring but probably worth.

Reply:
I flattened it out a bit but still wrote it in a way the func doesn't grow and grow with all the implementation. Let me know what you think.

Reply:
Definitely an improvement, but left a comment here: https://github.com/grafana/agent/pull/3998/files#r1210837896